File: case02/java/Prateleira.java
public class Prateleira
{
    private Produto produto1, produto2, produto3;
    
    public Prateleira(Produto produto1, Produto produto2, Produto produto3){
        this.produto1 = produto1;
        this.produto2 = produto2;
        this.produto3 = produto3;
    }
    
    public Prateleira(){};
    
    public Produto getProduto1(){
        return produto1;
    }
    public Produto getProduto2(){
        return produto2;
    }
    public Produto getProduto3(){
        return produto3;
    }
    public void setProduto1(Produto produto1){
        this.produto1 = produto1;
    }
    public void setProduto2(Produto produto2){
        this.produto2 = produto2;
    }
    public void setProduto3(Produto produto3){
        this.produto3 = produto3;
    }
    public Produto produtoMaisCaro(){
        Produto maisCaro = produto1;
        if (produto1.getPreco() < produto2.getPreco()){
            maisCaro = produto2;
            if (produto2.getPreco() < produto3.getPreco()){
                maisCaro = produto3;
            }
        }else if (produto1.getPreco() < produto3.getPreco()){
            maisCaro = produto3;
            if(produto2.getPreco() < produto3.getPreco()){
            }
        }
        return maisCaro;
    }
    public String toString(){
        return "Produto 1:\n" +produto1.toString()+ "\n\nProduto 2:\n" +produto2.toString() + "\n\nProduto 3:\n" +produto3.toString();
    }
}

File: case02/python/Prateleira.py
from Produto import Produto

class Prateleira:
    def __init__(self,prod1:Produto,prod2:Produto,prod3:Produto):
        self.prod1 = prod1
        self.prod2 = prod2
        self.prod3 = prod3
    
    def get_prod1(self):
        return self.prod1
    
    def get_prod2(self):
        return self.prod2
    
    def get_prod3(self):
        return self.prod3
    
    def set_prod1(self, prod1):
        self.prod1 = prod1

    def set_prod2(self, prod2):
        self.prod2 = prod2
    
    def set_prod3(self, prod3):
        self.prod3 = prod3

    def __str__(self):
        return f"Produto 1 - {self.prod1.get_nome()}\nProduto 2 - {self.prod2.get_nome()}\nProduto 3 - {self.prod3.get_nome()}"
    
    def produtoMaisCaro(self):
        maiscaro = self.prod1
        if self.prod1.get_preco() < self.prod2.get_preco():
            maisCaro = self.prod2
            if self.prod2.get_preco() < self.prod3.get_preco():
                maisCaro = self.prod3
        elif self.prod1.get_preco() < self.prod3.get_preco():
            maisCaro = self.prod3
        return maisCaro